logical/framework/openapi.go

```diff
@@ -0,0 +1,369 @@
+package framework
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/helper/openapi"
+	"github.com/hashicorp/vault/helper/wrapping"
+	"github.com/hashicorp/vault/logical"
+	"github.com/mitchellh/mapstructure"
+)
+
+// Regex for handling optional and named parameters, and string cleanup
+var optRe = regexp.MustCompile(`(?U)\(.*\)\?`)
```

Comment:
It'd be great to provide a brief comment on each of these that explains the rule out

Reply:
Will do.